public/app/plugins/datasource/mssql/configuration/ConfigurationEditor.tsx

```diff
@@ -0,0 +1,262 @@
+import { css } from '@emotion/css';
+import React, { SyntheticEvent } from 'react';
+
+import {
+  DataSourcePluginOptionsEditorProps,
+  GrafanaTheme2,
+  onUpdateDatasourceJsonDataOption,
+  onUpdateDatasourceSecureJsonDataOption,
+  SelectableValue,
+  updateDatasourcePluginJsonDataOption,
+  updateDatasourcePluginResetOption,
+} from '@grafana/data';
+import {
+  Alert,
+  FieldSet,
+  InlineField,
+  InlineFieldRow,
+  InlineSwitch,
+  Input,
+  SecretInput,
+  Select,
+  useStyles2,
+} from '@grafana/ui';
+import { ConnectionLimits } from 'app/features/plugins/sql/components/configuration/ConnectionLimits';
+
+import { MSSQLAuthenticationType, MSSQLEncryptOptions, MssqlOptions } from '../types';
+
+export const ConfigurationEditor = (props: DataSourcePluginOptionsEditorProps<MssqlOptions>) => {
+  const { options, onOptionsChange } = props;
+  const styles = useStyles2(getStyles);
+  const jsonData = options.jsonData;
+
+  const onResetPassword = () => {
+    updateDatasourcePluginResetOption(props, 'password');
+  };
+
+  const onDSOptionChanged = (property: keyof MssqlOptions) => {
+    return (event: SyntheticEvent<HTMLInputElement>) => {
+      onOptionsChange({ ...options, ...{ [property]: event.currentTarget.value } });
+    };
+  };
+
+  const onSkipTLSVerifyChanged = (event: SyntheticEvent<HTMLInputElement>) => {
+    updateDatasourcePluginJsonDataOption(props, 'tlsSkipVerify', event.currentTarget.checked);
+  };
+
+  const onEncryptChanged = (value: SelectableValue) => {
+    updateDatasourcePluginJsonDataOption(props, 'encrypt', value.value);
+  };
+
+  const onAuthenticationMethodChanged = (value: SelectableValue) => {
+    onOptionsChange({
+      ...options,
+      ...{
+        jsonData: { ...jsonData, ...{ authenticationType: value.value } },
+        secureJsonData: { ...options.secureJsonData, ...{ password: '' } },
+        secureJsonFields: { ...options.secureJsonFields, ...{ password: false } },
+        user: '',
+      },
+    });
+  };
+
+  const authenticationOptions: Array<SelectableValue<MSSQLAuthenticationType>> = [
+    { value: MSSQLAuthenticationType.sqlAuth, label: 'SQL Server Authentication' },
+    { value: MSSQLAuthenticationType.windowsAuth, label: 'Windows Authentication' },
+  ];
+
+  const encryptOptions: Array<SelectableValue<string>> = [
+    { value: MSSQLEncryptOptions.disable, label: 'disable' },
+    { value: MSSQLEncryptOptions.false, label: 'false' },
+    { value: MSSQLEncryptOptions.true, label: 'true' },
+  ];
+
+  const shortWidth = 15;
+  const longWidth = 46;
+  const labelWidthSSL = 25;
+
+  return (
+    <>
+      <FieldSet label="MS SQL Connection" width={400}>
+        <InlineField labelWidth={shortWidth} label="Host">
+          <Input
+            width={longWidth}
+            name="host"
+            type="text"
+            value={options.url || ''}
+            placeholder="localhost:1433"
+            onChange={onDSOptionChanged('url')}
+          ></Input>
```

Comment:
It would be cleaner to use self-closing tag. It is occurring in multiple input tags here.

Reply:
I agree with that it would be better.

But I would avoid changes the code in the mssql directory as it's pure grafana.

https://github.com/grafana/grafana/blob/release-9.2.5/public/app/plugins/datasource/mssql/configuration/ConfigurationEditor.tsx

Initially I've merged it wrong and had to add that in after the fact.